File: PHP/Transactions/Check_Customer_Credit.php
<?php
/**
* BluePay PHP Sample Code
*
* This code sample runs a $3.00 Credit Card Auth transaction
* against a customer using test payment information.
* If using TEST mode, odd dollar amounts will return
* an approval and even dollar amounts will return a decline.
*/

include('../BluePay.php');

$accountID = "Merchant's Account ID Here";
$secretKey = "Merchant's Secret Key Here";
$mode = "TEST";

$payment = new BluePay(
    $accountID,
    $secretKey,
    $mode
);

$payment->setCustomerInformation(array(
    'firstName' => 'Bob', 
    'lastName' => 'Tester', 
    'addr1' => '1234 Test St.', 
    'addr2' => 'Apt #500', 
    'city' => 'Testville', 
    'state' => 'IL', 
    'zip' =>'54321', 
    'country' => 'USA', 
    'phone' => '1231231234', 
    'email' => 'test@bluepay.com' 
));

$payment->setCCInformation(array(
    'cardNumber' => '4111111111111111', // Card Number: 4111111111111111
    'cardExpire' => '1225', // Card Expire: 12/25
    'cvv2' => '123' // Card CVV2: 123
));


$payment->auth('3.00'); // Card authorization amount: $3.00

// Makes the API Request for a credit card authorization
$payment->process(); 

if($payment->isSuccessfulResponse()){
# Read response from BluePay
    echo 
    'Transaction Status: '. $payment->getStatus() . "\n" .
    'Transaction Message: '. $payment->getMessage() . "\n" .
    'Transaction ID: '. $payment->getTransID() . "\n" .
    'AVS Response: ' . $payment->getAVSResponse() . "\n" .
    'CVS Response: ' . $payment->getCVV2Response() . "\n" .
    'Masked Account: ' . $payment->getMaskedAccount() . "\n" .
    'Card Type: ' . $payment->getCardType() . "\n" .
    'Authorization Code: ' . $payment->getAuthCode() . "\n";
} else{
    echo $payment->getMessage() . "\n";
}
?>

File: C++/Transactions/Check_Customer_Credit.cpp
//
// BluePay C++ Sample code.
//
// This code sample runs a $3.00 Credit Card Authorization transaction
// against a customer using test payment information.  
// If using TEST mode, odd dollar amounts will return
// an approval and even dollar amounts will return a decline.

#include "Check_Customer_Credit.h"
#include "../bluepay-cpp/BluePay.h"
#include <iostream>
using namespace std;

void checkCustomerCredit(){

  string accountId = "Merchant's Account ID Here";
  string secretKey = "Merchant's Secret Key Here";
  string mode = "TEST";

	BluePay payment(
		accountId, 
		secretKey, 
		mode 
	);

  payment.setCustomerInformation(
    "Bob", // First Name
    "Tester", // Last Name
    "123 Test St.", // Address1
    "Apt #500", // Address2
    "Testville", // City
    "IL", // State
    "54321", // Zip
    "USA", // Country
    "1231231234", // Phone Number
    "test@bluepay.com" // Email Address
  );

  payment.setCCInformation(
    "4111111111111111", // Card Number
    "1225", // Card Expire
    "123" // Card CVV2
  );

	// Auth Amount: $3.00
	payment.auth("3.00");

	// Makes the API Request with Blue
	payment.process();

  // Reads the responses from BluePay if transaction was approved
  if (payment.isSuccessfulTransaction()){
    cout << string("Transaction Status: ") + payment.getResult() + "\n";
    cout << string("Transaction Message: ") + payment.getMessage() + "\n";
    cout << string("Transaction ID: ") + payment.getTransId() + "\n";
    cout << string("AVS Result: ") + payment.getAvs() + "\n";
    cout << string("CVV2 Result: ") + payment.getCvv2() + "\n";
    cout << string("Masked Payment Account: ") + payment.getMaskedPaymentAccount() + "\n";
    cout << string("Card Type: ") + payment.getCardType() + "\n";
    cout << string("Authorization Code: ") + payment.getAuthCode() + "\n";
  }
  else {
      cout << string("Error: ") + payment.getMessage();
  }

}